src/_pytest/config/__init__.py

```diff
@@ -1104,10 +1108,14 @@ def inipath(self) -> pathlib.Path | None:
         """
         return self._inipath
 
-    def add_cleanup(self, func: Callable[[], None]) -> None:
+    def add_cleanup(self, func: _T_callback) -> _T_callback:
         """Add a function to be called when the config object gets out of
-        use (usually coinciding with pytest_unconfigure)."""
-        self._cleanup.append(func)
+        use (usually coinciding with pytest_unconfigure).
+
+        Returns the passed function.
```

Comment:
```suggestion
        Returns the passed function, so you can use @config.add_cleanup as a decorator
```

Reply:
Oh I see, thanks.

However is that useful in production code? I can only see this being used in tests, given it needs a config object to use as decorator?

Reply:
I think it's probably more useful generally this way

Reply:
Hmm unless we have authentic uses for this in production, I feel we will be committing to this API which might be a problem in the future in case we want to evolve it and return something else (just an example, we could return an object which can be used to remove the callback from the cleanup stack, for example), for the small convenience of being able to use it as a decorator.

Unless you feel really strong about it, can you revert the return type to `None`? Thanks.